### PR TITLE
Add AITER backend support for Flashinfer SinglePrefill

### PR DESCRIPTION
**Pull request by rtmadduri**
This PR integrates experimental support of AITER backend as an alternative to FA2 for Single Prefill operations with paged KV cache

Key Changes

**Key Changes**

- AITER Backend Detection & Setup
  - Added runtime detection for AITER availability on ROCm systems
  - Introduced `HAS_AITER` flag and aiter_mha_module import
  - Fallback to FA2 backend when AITER is not installed

- AITER Backend Implementation
  - `_get_aiter_single_prefill_module()`: Module-like namespace providing uniform interface with FA2
  - `aiter_paged_run()`: Main execution function for paged KV cache with AITER backend
  - `_aiter_noop_plan()`: No-op planning function (AITER doesn't require separate planning)

- Features
  - Full paged KV cache support
  - Causal masking and window attention support
  - Logits soft-capping support

- Test Coverage
  - Added unit tests to compare AITER single prefill calls with FA2 single prefill calls
  - Parametrized tests to run with both "fa2" and "aiter" backends
  - Added backend-specific test skipping (e.g., AITER with causal mode or non-NHD layouts)
  - Extended `test_single_prefill_with_kv_cache()`

Unit Test Results
* **test_single_prefill_kernels_hip.py**
```
===================================== 1444 passed, 480 skipped in 6.35s =====================================
```
* **Complete ROCm Flashinfer Test Suite**
```
=====================================  26036 passed, 2841 skipped in 426.81s (0:07:06) ===================================== 
```
To test:

Install pandas - dependency for AITER
```
pip install pandas
```

Install AITER

```
git clone --recursive https://github.com/ROCm/aiter.git
cd aiter
python setup.py develop
```

JIT Install Flashinfer
```
FLASHINFER_HIP_ARCHITECTURES=gfx942 \
  python -m pip wheel . --wheel-dir=./dist/ --no-deps --no-build-isolation -v
cd dist && pip install amd_flashinfer-*.whl
```

Run Tests
```
pytest tests/test_single_prefill_kernels_hip.py
```
Full Test Suite - From the Flashinfer src dir:
```
pytest
```

### CODE REVIEW

File: tests/test_single_prefill_kernels_hip.py [hunk 1]
```diff
@@ -89,7 +98,7 @@ def test_single_prefill_with_kv_cache(
             kv_layout=kv_layout,
             pos_encoding_mode=pos_encoding_mode,
             logits_soft_cap=logits_soft_cap,
-            backend="fa2",
+            backend=backend,
```
Comment (diptorupd) [inline tests/test_single_prefill_kernels_hip.py:101]: 👍

**Review by diptorupd — APPROVED** [review]
LGTM!
